### DISCUSSION TITLE
cloudflare protection / error

### DISCUSSION THREAD

**Discussion by c4ir0** [projectdiscovery/httpx#884] [General] [answered]
#### I test a site which I want to probe urls using httpx. However, the site I'm talking about is using cloudflare protection, after requesting => it says 'checking your browser, you will be redirected in 5 seconds' and I'm getting redirected to the page that doesn't exists because as I think I didn't pass some protection test [always returns 403 for my requests], after some search I knew That it would be "CloudFlare's I'm Under Attack Mode" --> http://blog.cloudflare.com/introducing-im-under-attack-mode 

![404_](https://user-images.githubusercontent.com/108268587/202917333-aed0a827-c4c2-4d53-9f89-8597adc13f81.JPG)

>**I run the same probe to this website using cURl and also have the same results but when I run the command as "copy as curl command from BurpSuite options" it returned the right status codes like 401, 404,200, ...** 

![command](https://user-images.githubusercontent.com/108268587/202917426-4f13c5dd-c7ee-4351-93cb-3e1ba6d3e0aa.png)

>**So I copied the same headers from the last cURL command and used in httpx but still the same [ always 403 ]** 

![after adding H](https://user-images.githubusercontent.com/108268587/202917594-864ed22c-a934-40d8-a6fc-d25a678238dc.JPG)

## regards

**Comment by ehsandeep** [comment, marked as answer]
@c4ir0 this is expected with cloudflare protection, for these cases, you probably going to require a headless request for automation.

**Reply by c4ir0** [reply]
Kindly, can u give a reference or short brief how to do so + I thought about that to have headless browser but didn't find an option in httpx like katana

**Reply by ehsandeep** [reply]
That's not available/possible with httpx.

**Reply by c4ir0** [reply]
> That's not available/possible with httpx.

If there another method could you help me, bro 🥺
Just give me the key to start and I'll do the rest ❤️

**Reply by olearycrew** [reply]
Using nuclei or another tool that allows headless requests should help.
